R/knitr-engine.R

```diff
@@ -492,19 +492,44 @@ eng_python_initialize_hooks <- function(options, envir) {
 
 }
 
+eng_python_matplotlib_backend <- function() {
+
+  # allow override, just in case
+  envvars <- c("RETICULATE_MPLBACKEND", "MPLBACKEND")
+  for (envvar in envvars) {
+    override <- Sys.getenv(envvar, unset = NA)
+    if (!is.na(override))
+      return(override)
+  }
+
+  # in RStudio Desktop, enforce a non-Qt matplotlib backend
+  # this is especially important with RStudio Desktop as attempting to use a Qt
+  # backend will cause issues due to mismatched Qt versions between RStudio and
+  # Anaconda environments, and will cause crashes when generating plots
```

Comment:
Is this a conda specific workaround? Would a venv python installation find the same Qt version as the IDE? If not, should the comment be updated?

Reply:
IIRC, this branch was specifically required for older versions of RStudio (which were built against Qt); I've updated the comment to indicate that and also note that we could probably remove this in a future `reticulate` release. Does that seem sensible?